vm/devices/vmbus/vmbus_server/src/lib.rs

```diff
@@ -1324,11 +1381,44 @@ impl channels::Notifier for ServerTaskInner {
         }
     }
 
-    fn send_message(&mut self, message: OutgoingMessage, target: Option<ConnectionTarget>) {
-        let ConnectionTarget { vp, sint } = target.unwrap_or(self.message_target);
+    fn send_message(&mut self, message: OutgoingMessage, target: MessageTarget) {
+        let port = match target {
+            MessageTarget::Default => util::BorrowedOrOwned::Borrowed(&self.message_port),
+            MessageTarget::ReservedChannel(offer_id) => {
+                let channel = self
+                    .channels
+                    .get(&offer_id)
+                    .expect("channel does not exist");
+                let message_port = match &channel.state {
+                    ChannelState::Open {
+                        reserved_guest_message_port: Some(message_port),
+                        ..
+                    }
+                    | ChannelState::ClosingReserved(message_port) => message_port,
+                    _ => unreachable!("channel is not reserved"),
+                };
+                util::BorrowedOrOwned::Borrowed(message_port)
+            }
+            MessageTarget::Custom(target) => util::BorrowedOrOwned::Owned(
```

Comment:
The more common way to handle this pattern, without a custom type, is something like this:

```rust
let foo_storage;
let foo = if let Some(borrowed_foo) = &self.maybe_foo {
    borrowed_foo
} else {
    foo_storage = Foo::new();
    &foo_storage
};
```